projects/core/src/occ/adapters/product/converters/product-image-normalizer.ts

```diff
@@ -54,10 +54,13 @@ export class ProductImageNormalizer implements Converter<Occ.Product, Product> {
          * `backend.media.baseUrl` by default, but fallback to `backend.occ.baseUrl`
          * if none provided.
          */
-        image.url =
-          (this.config.backend.media.baseUrl ||
-            this.config.backend.occ.baseUrl ||
-            '') + image.url;
+        image.url = image.url.startsWith('http')
```

Comment:
@kbalagtey-tacit sorry for the slow review here. I like this idea very much. 
Would you be comfortable to add a unit test for this feature? 
You could also move this in a separate method for readability and avoiding such a long method (not your bad).

Reply:
@kbalagtey-tacit let us know if you're willing to continue here